src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -0,0 +1,191 @@
+
+  override protected val invokeTextAnalytics: String => TAResponseV4[DetectedLanguageV4] = (text: String) =>
+    {
+      val detectLanguageResultCollection = textAnalyticsClient.detectLanguageBatch(
+        Seq(text).asJava, null, textAnalyticsOptions.orNull)
+      val detectLanguageResult = detectLanguageResultCollection.asScala.head
+
+      val languageResult = if (detectLanguageResult.isError) {
+        None
+      } else {
+        Some(DetectedLanguageV4(
+          detectLanguageResult.getPrimaryLanguage.getName,
+          detectLanguageResult.getPrimaryLanguage.getIso6391Name,
+          detectLanguageResult.getPrimaryLanguage.getConfidenceScore))
+      }
+
+      val error = if (detectLanguageResult.isError) {
+        val error = detectLanguageResult.getError
+        Some(TAErrorV4(error.getErrorCode.toString, error.getMessage, error.getTarget))
+      } else {
+        None
+      }
+
+      val stats = Option(detectLanguageResult.getStatistics) match {
+        case Some(s) => Some(DocumentStatistics(s.getCharacterCount, s.getTransactionCount))
+        case None => None
+      }
+
+      TAResponseV4[DetectedLanguageV4](
+        languageResult,
+        error,
+        stats,
+        Some(detectLanguageResultCollection.getModelVersion))
+    }
+}
+
+object DetectLanguageResponseV4 extends SparkBindings[TAResponseV4[DetectedLanguageV4]]
+object KeyPhraseResponseV4 extends SparkBindings[TAResponseV4[KeyphraseV4]]
+
+case class TAResponseV4[T](result: Option[T],
+                           error: Option[TAErrorV4],
+                           statistics: Option[DocumentStatistics],
+                           modelVersion: Option[String])
+
+case class TAErrorV4(errorCode: String, errorMessage: String, target: String)
+
+case class DetectedLanguageV4(name: String, iso6391Name: String, confidenceScore: Double)
+
+object TextAnalyticsKeyphraseExtraction extends ComplexParamsReadable[TextAnalyticsKeyphraseExtraction]
+
+class TextAnalyticsKeyphraseExtraction (override val textAnalyticsOptions: Option[TextAnalyticsRequestOptions] = None,
+                                     override val uid: String = randomUID("TextAnalyticsKeyphraseExtraction"))
+  extends TextAnalyticsSDKBase[KeyphraseV4](textAnalyticsOptions) {
+  logClass()
+
+  override def outputSchema: StructType =  KeyPhraseResponseV4.schema
+
+  override protected val invokeTextAnalytics: String => TAResponseV4[KeyphraseV4] = (text: String) =>
+  {
+    val ExtractKeyPhrasesResultCollection = textAnalyticsClient.extractKeyPhrasesBatch(
+      Seq(text).asJava,"en", textAnalyticsOptions.orNull)
+    val keyPhraseExtractionResult = ExtractKeyPhrasesResultCollection.asScala.head
+    val keyPhraseDocument = keyPhraseExtractionResult.getKeyPhrases()
+
+    val keyphraseResult = if (keyPhraseExtractionResult.isError) {
+      None
+    } else {
+      Some(KeyphraseV4(
+        keyPhraseDocument.asScala.toList,
+        keyPhraseDocument.getWarnings.asScala.toList.map(

```

Comment:
I think you want the `toList` after the map, otherwise it won't actually instantiate straight away (will be reevaluate on every access)